db/fetcher/indexer.go

```diff
@@ -54,14 +60,18 @@ func NewIndexFetcher(
 
 func (f *IndexFetcher) Init(
 	ctx context.Context,
+	identity immutable.Option[string],
 	txn datastore.Txn,
+	acp immutable.Option[acp.ACPModule],
 	col client.Collection,
 	fields []client.FieldDefinition,
 	filter *mapper.Filter,
 	docMapper *core.DocumentMapping,
 	reverse bool,
 	showDeleted bool,
 ) error {
+	f.identity = identity
+	f.acp = acp
```

Comment:
todo: Similar to the lens fetcher, there appears to be no reason for these to be on the IndexFetcher struct - please remove this extra state

Reply:
You would rather I completely remove these even the function param needs to be there? I think this would assist whoever is implementing the index and lens fetcher (as this might not be so obvious to trace).

What do you say if I left a Todo marked with an issue with these?

Reply:
> You would rather I completely remove these even the function param needs to be there

Yes, there is no benefit to having them on there, and less state (and code) is better than more.

> I think this would assist whoever is implementing the index and lens fetcher (as this might not be so obvious to trace).

I do not understand what you mean here.  Index/LensFetchter.Init just receives them as params and passes them immediately into docFetcher.Init, they are not needed anywhere else but in this function.

Reply:
Ah I see, you are saying to pass it directly to the args that need it and remove from struct.

- [ ] Will do